cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -1201,4 +1316,29 @@ private  String decodeUserId(String encodedUserId) {
     }
     return decodedUserId;
   }
+
+  private void checkLatestVersionExeceution(ProgramId programId)
+    throws BadRequestException, ApplicationNotFoundException {
+    String latestVersion = getLatestApplicationId(programId.getNamespaceId(), programId.getApplication()).getVersion();
+    if (!latestVersion.equals(programId.getVersion())) {
+      throw new BadRequestException(String.format("Start action is only allowed on the latest version %s",
```

Comment:
Let's also mention that the versionless api should be used instead. Something like

"... version %s. Please use the versionless start API instead."